### PR TITLE
Edx 123 display api key message on code blocks

### PR DESCRIPTION
**Pull request by MymmiJ**
> _NOTE TO REVIEWERS_ - please **do not review PRs in the `DRAFT` state**, as the PR may change substantially before it is ready to review. Thanks.

## Description

This PR covers the following ground:
* [EDX-123: Inconsistent display of API keys in legacy toolchain](https://ably.atlassian.net/browse/EDX-123).
* [EDX-48: Enable Ably login via. Meganav in Gatsby toolchain](https://ably.atlassian.net/browse/EDX-48)
* [EDX-49: Render API keys](https://ably.atlassian.net/browse/EDX-49)
EDX-123 is covered by our ability to treat all Code elements the same in the new toolchain.
EDX-48 has been covered en-route to sorting out desired API key behaviour
EDX-49 has been accomplished by:
* Where there is an {{API_KEY}} token in a \<Code\> element
* If we are signed in, render a series of asterisks instead of the token but allow the user API key to be copied by clicking the 'Copy' button.
* If we are not signed in, render a temporary demo API key in full instead of the token.

## Review

Instructions on how to review the PR. 

* [Page to review](http://localhost:8000/docs/realtime/connection) (gatsby develop)
You should see an API key indicator that provides you with a simple tooltip on mouseover.

To view the API key dropdown, enter the following data in a `.env.development` file:

```
GATSBY_DOCS_SIGNED_IN='true'
GATSBY_DOCS_API_KEYS='true'
```

Then run `gatsby clean && gatsby develop`.

To view 'real' (dev) data in the API key dropdown menu:

* Run this website branch: WEB-2263-proxy-new-docs-toolchain
* View the website PR here for changes, if it is merged simply run website/main: https://github.com/ably/website/pull/4424
* Run `gatsby clean && gatsby build --prefix-paths && gatsby serve --prefix-paths`

* [Page to review](http://localhost:3000/docs/realtime/connection) (running subordinate to website app)

## Minor Fixes

Please ask if you want any of these fixes or tweaks to be removed, which are not related to the primary purposes of the PR:

1. Language buttons display the raw lang value (e.g. 'sh') instead of nothing (e.g. '') when the language doesn't have a relevant 'label' (e.g. 'Shell Script' or 'Shell').
2. README has been split into LEGACY_README and README for purposes of maintainability and readability
3. tabIndex changed to {0} rather than incorrect type "0" in <CopyLink> wrapper class
4. 404 page moved into a location where it can be accessed on the site, /docs/404 rather than just /404

### CODE REVIEW

File: src/components/blocks/software/Code/CodeCopyButton.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React, { Dispatch, useEffect, useState } from 'react';
+import '@ably/ui/core/styles.css';
+import { gui } from '../../../../styles/colors';
+
+const DEFAULT_BUTTON_CONTENT = `Copy ⎘`;
+
+const CodeCopyButton = ({ content }: { content: string }) => {
+  const [buttonContent, setButtonContent]: [string, Dispatch<React.SetStateAction<string>>] =
```
Comment (DuncanEdwards) [inline src/components/blocks/software/Code/CodeCopyButton.tsx:8]: No need for explicit types here.

File: src/components/blocks/software/Code/CodeCopyButton.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    useState<string>(DEFAULT_BUTTON_CONTENT);
+  const [copySuccess, setCopySuccess]: [boolean | null, Dispatch<React.SetStateAction<boolean | null>>] = useState<
```
Comment (DuncanEdwards) [inline src/components/blocks/software/Code/CodeCopyButton.tsx:10]: Or here

File: src/components/blocks/software/Code/CodeCopyButton.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React, { Dispatch, useEffect, useState } from 'react';
```
Comment (DuncanEdwards) [inline src/components/blocks/software/Code/CodeCopyButton.tsx:1]: Dispatch unused

File: src/redux/api-key/api-key-reducer.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  url: string;
+};
+
+type ApiKeyState = {
+  data: Array<Record<string, ApiKeyValue>>;
+};
+
+type Action = {
+  payload: Record<string, unknown>;
+  type: string;
```
Comment (DuncanEdwards) [inline src/redux/api-key/api-key-reducer.ts:14]: You can use restrict this type further to restrict to only the supported actions

**Review by DuncanEdwards — CHANGES_REQUESTED** [review]
Only minor changes

File: src/components/blocks/software/Code/CodeCopyButton.tsx (hunk 1 above)
Comment (MymmiJ) [reply src/components/blocks/software/Code/CodeCopyButton.tsx]: Removed & tested.

File: src/components/blocks/software/Code/CodeCopyButton.tsx (hunk 2 above)
Comment (MymmiJ) [reply src/components/blocks/software/Code/CodeCopyButton.tsx]: Removed & tested.

File: src/redux/api-key/api-key-reducer.ts (hunk 4 above)
Comment (MymmiJ) [reply src/redux/api-key/api-key-reducer.ts]: Restricted; currently only one supported action.

**Review by DuncanEdwards — APPROVED** [review]
GO GO GO!

**Pull request merged**